t/08_session/03_http_requests.t

```diff
@@ -55,6 +55,22 @@ Test::TCP::test_tcp(
             like $res->{content}, qr/name='$client'/,
             "session looks good for client $client";
 
+            $res = $ua->get("http://127.0.0.1:$port/session/after_hook/read");
+            ok($res->{success}, "Reading a session var in after hook worked");
```

Comment:
You didn't try to write a session variable in that hook. You're testing the hook where you just read the session.